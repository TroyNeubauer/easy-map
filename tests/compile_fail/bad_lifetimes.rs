fn main() {
    let r = {
        let map = easy_map::EasyMap::<usize, usize>::new();
        map.insert(0, 0);
        map.get(&0)
    };
    assert_eq!(r.as_deref(), Some(&0));
}
