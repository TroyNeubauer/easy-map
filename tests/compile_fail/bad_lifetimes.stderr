error[E0597]: `map` does not live long enough
 --> tests/compile_fail/bad_lifetimes.rs:5:9
  |
2 |     let r = {
  |         - borrow later stored here
...
5 |         map.get(&0)
  |         ^^^^^^^^^^^ borrowed value does not live long enough
6 |     };
  |     - `map` dropped here while still borrowed
